### ISSUE TITLE
Digit parser takes in empty characters.

### ISSUE THREAD

**Issue by wshatch** [Geal/nom#142]
While working with the digit parser and chain I came across this issue
```rust
named!(parse_ints< Vec<i32> >, many0!(spaces_or_int));

named!(spaces_or_int<&[u8], i32>, chain!(
    opt!(space) ~
    int: digit,
    || {
      let result = str::from_utf8(int).unwrap();
      println!("Result: {}", result);
      println!("int is empty?: {}", int.is_empty());
      match result.parse(){
        Ok(i) => i,
        Err(_) =>  panic!("UH OH! NOT A DIGIT!")
      }
    }
));

#[test]
fn test_parse_ints(){
   let subject = parse_ints(&b"12 34 5689"[..]);
   let expected = IResult::Done(&b""[..], vec![12, 34, 5689]);
   assert_eq!(subject, expected)
}
```
outputs

```rust
test nom_parse_funcs::test_parse_ints ... FAILED

failures:

---- nom_parse_funcs::test_parse_ints stdout ----
        Result: 12
int is empty?: false
Result: 34
int is empty?: false

Result: 5689
int is empty?: false
Result:
int is empty?: true
thread 'nom_parse_funcs::test_parse_ints' panicked at 'UH OH! NOT A DIGIT!', src/nom_parse_funcs.rs:14
```

**Comment by wshatch** [comment]
So after further testing, this doesn't appear to happen in the master branch, just in 1.0.1.

**Comment by wshatch** [comment]
Even after further testing, this issue comes up if you parse an optional parser in chain at the end. For example,
```rust
 let subject = parse_ints(&b"12 34 5689 "[..]);
```
for parse ints.

**Comment by Geal** [comment]
Hi!

Thank you for your report. I just fixed it in 2eddc3d, it will appear in the 1.1 release.
